### ISSUE TITLE
GRPO训练中Loss和grad_norm一直为0

### ISSUE THREAD

**Issue by Rinnolo** [modelscope/ms-swift#4570]
训练脚本如下，可以看到除了改了一些训练设置之外几乎都为example文件夹中的内容：

```
MAX_PIXELS=602112 \
WANDB_API_KEY=xxx \
CUDA_VISIBLE_DEVICES=0,1 \
NPROC_PER_NODE=2 \
swift rlhf \
  --rlhf_type grpo \
  --model /mdata/llh/Models/Qwen2.5-VL-3B-Instruct \
  --train_type lora \
  --use_vllm true \
  --vllm_mode colocate \
  --vllm_gpu_memory_utilization 0.6 \
  --vllm_max_model_len 8192 \
  --vllm_tensor_parallel_size 2 \
  --dataset /mdata/llh/Datasets/multimodal-open-r1-8k-verified#1000 \
  --external_plugins /mdata/llh/Projects/ms-swift/examples/train/grpo/plugin/plugin.py \
  --reward_funcs external_r1v_acc format \
  --reward_weights 1 0.1 \
  --torch_dtype bfloat16 \
  --attn_impl flash_attn \
  --num_train_epochs 1 \
  --max_length 8192 \
  --per_device_train_batch_size 4 \
  --per_device_eval_batch_size 2 \
  --gradient_accumulation_steps 1 \
  --eval_steps 500 \
  --save_steps 500 \
  --learning_rate 1e-6 \
  --save_total_limit 2 \
  --logging_steps 1 \
  --warmup_ratio 0.05 \
  --dataloader_num_workers 2 \
  --max_completion_length 2048 \
  --num_generations 8 \
  --deepspeed zero3 \
  --temperature 1.1 \
  --top_p 1.0 \
  --top_k 80 \
  --log_completions true \
  --async_generate false \
  --offload_optimizer true \
  --offload_model true \
  --gc_collect_after_offload true \
  --move_model_batches 40 \
  --sleep_level 1 \
  --report_to swanlab \
  --swanlab_project qwen2.5vl_3b_test \
  --system examples/train/grpo/prompt.txt
```

**Comment by hjh0119** [comment]
1. Check if the Swift version is >= 3.5.
2. Check if reward_std is 0 during training.
3. If reward_std is not 0, attempt to upgrade DeepSpeed(>=0.16.9)

**Comment by Rinnolo** [comment]
I have check the Swift version and Deepspeed Version, which is 3.5.0.dev0 and 0.17.0.

<img width="760" alt="Image" src="https://github.com/user-attachments/assets/844835e6-be60-44c4-8010-20382c5defbb" />

I am sure that reward_std is 0 during training,you can visit it through the image i upload.

<img width="1152" alt="Image" src="https://github.com/user-attachments/assets/aba809c6-859d-461b-b270-cbb3b5b483d7" />

**Comment by hjh0119** [comment]
There may be bugs in version 3.5.0.dev0. Please update the code and try again
```
# 3.5 release
pip install ms-swift -U
```

or
```
# 3.6.dev
cd /path/to/ms-swift
git pull
pip install -e .
```

**Comment by Rinnolo** [comment]
> There may be bugs in version 3.5.0.dev0. Please update the code and try again
> 
> ```
> # 3.5 release
> pip install ms-swift -U
> ```
> 
> or
> 
> ```
> # 3.6.dev
> cd /path/to/ms-swift
> git pull
> pip install -e .
> ```

ok, I try 3.5 release first. 
It may need 1 day to validate at least(or more?).Just do not close this issue, I would update it later if problem still exists.

**Comment by chaunceyliu30** [comment]
I had the same problem yesterday, it might have something to do with your reward function. In my case, it's because my completions include think content and I use '==' to calculate reward, it caused the reward always stay at 0. Hope this helps.

**Comment by hjh0119** [comment]
any update?

**Comment by dma2077** [comment]
I encountered the same problem. 

<img width="1591" alt="Image" src="https://github.com/user-attachments/assets/43b313fc-6481-4304-b004-a3841054f56b" />

My reward/std also looked abnormal, its value was 0, but my reward was composed of accuracy and format, accuracy/std was greater than 0, but the final reward/std was 0.
![Image](https://github.com/user-attachments/assets/af46566a-d7f1-4e0a-9ceb-00045332a3d3)

**Comment by hjh0119** [comment]
@dma2077 what's the version of swift/vllm/transformers?

how to repro?

**Comment by xjy2020** [comment]
我训练时format的reward一直显示为0

![Image](https://github.com/user-attachments/assets/55d39b79-6102-487d-931c-c136adeb2cee)

训练脚本如下

export CUDA_VISIBLE_DEVICES=0,1,2,3
export NPROC_PER_NODE=4
export WANDB_DISABLED=true
export WANDB_MODE=disabled

swift rlhf \
  --rlhf_type grpo \
  --model /data/models/bluelm \
  --model_type llama \
  --external_plugins examples/train/grpo/plugin/plugin.py \
  --reward_funcs external_countdown external_math_format \
  --reward_weights 1 0.1 \
  --train_type full \
  --use_vllm true \
  --vllm_device auto \
  --vllm_gpu_memory_utilization 0.5 \
  --vllm_max_model_len 2048 \
  --torch_dtype bfloat16 \
  --dataset '/data/datasets/Countdown-Tasks-3to4#30000' \
  --max_completion_length 1024 \
  --num_train_epochs 1 \
  --per_device_train_batch_size 4 \
  --per_device_eval_batch_size 4 \
  --learning_rate 1e-6 \
  --gradient_accumulation_steps 4 \
  --eval_steps 500 \
  --save_steps 100 \
  --save_total_limit 4 \
  --logging_steps 5 \
  --max_length 2048 \
  --output_dir output/bluelm/full/countdown1 \
  --warmup_ratio 0.05 \
  --dataloader_num_workers 0 \
  --dataset_num_proc 4 \
  --num_generations 8 \
  --temperature 0.9 \
  --deepspeed zero3 \
  --system 'You are a helpful assistant. You first thinks about the reasoning process in the mind and then provides the user with the answer.' \
  --log_completions true \
  --num_iterations 1 \
  --beta 0.001

**Comment by hjh0119** [comment]
maybe fixed in https://github.com/modelscope/ms-swift/pull/4712

**Comment by ZhuoranZhao** [comment]
> I encountered the same problem.
> 
> <img alt="Image" width="1591" src="https://private-user-images.githubusercontent.com/76395131/458492442-43b313fc-6481-4304-b004-a3841054f56b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ1NzgzMTgsIm5iZiI6MTc1NDU3ODAxOCwicGF0aCI6Ii83NjM5NTEzMS80NTg0OTI0NDItNDNiMzEzZmMtNjQ4MS00MzA0LWIwMDQtYTM4NDEwNTRmNTZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA3VDE0NDY1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3OTUyNjcxNzg1ZGVkMzZjMzZmMDk1YmI4NTVlMDE2ZWQ0NzczNTI4ZmUxY2M2NDhhZmIxOTI1MGE5ZDVjOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xBUHmd59lXPXmZuqgPtnH3gzcNajtT91JAE40Syk_z0">
> My reward/std also looked abnormal, its value was 0, but my reward was composed of accuracy and format, accuracy/std was greater than 0, but the final reward/std was 0. ![Image](https://github.com/user-attachments/assets/af46566a-d7f1-4e0a-9ceb-00045332a3d3)

我也遇到了这种情况，你现在知道是为什么了吗？

**Comment by dma2077** [comment]
> > I encountered the same problem.
> > <img alt="Image" width="1591" src="https://private-user-images.githubusercontent.com/76395131/458492442-43b313fc-6481-4304-b004-a3841054f56b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ1NzgzMTgsIm5iZiI6MTc1NDU3ODAxOCwicGF0aCI6Ii83NjM5NTEzMS80NTg0OTI0NDItNDNiMzEzZmMtNjQ4MS00MzA0LWIwMDQtYTM4NDEwNTRmNTZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA3VDE0NDY1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3OTUyNjcxNzg1ZGVkMzZjMzZmMDk1YmI4NTVlMDE2ZWQ0NzczNTI4ZmUxY2M2NDhhZmIxOTI1MGE5ZDVjOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xBUHmd59lXPXmZuqgPtnH3gzcNajtT91JAE40Syk_z0">
> > My reward/std also looked abnormal, its value was 0, but my reward was composed of accuracy and format, accuracy/std was greater than 0, but the final reward/std was 0. ![Image](https://github.com/user-attachments/assets/af46566a-d7f1-4e0a-9ceb-00045332a3d3)
> 
> 我也遇到了这种情况，你现在知道是为什么了吗？

我不知道，我换了其他的框架

**Comment by hjh0119** [comment]
The two compute standard deviation (std) in different ways:

- The std of individual rewards is calculated over all the data (potentially across multiple groups).
- The overall reward std is computed only within each group.

Therefore, if you find that the std of individual rewards is nonzero while the overall reward std is zero, it probably means that the model's output randomness is too low, resulting in identical rewards within each group (i.e., for each prompt).

You can also use `log_completions` to check whether the model's outputs are abnormal.
